optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -243,6 +247,14 @@ public void setAssertionScoreDirectorFactory(ScoreDirectorFactoryConfig assertio
         this.assertionScoreDirectorFactory = assertionScoreDirectorFactory;
     }
 
+    public boolean isGenerateDroolsTestOnError() {
+        return generateDroolsTestOnError != null && generateDroolsTestOnError.booleanValue();
```

Comment:
The getter should return null if it's null.
For all Config classes, it's important to distinguish between null (= not filled in) and the default value (the value it gets when it's null). This for example impacts the programmatic API after reading a solver config XML file. Also note that solver config files are sometimes written to XML (see benchmark report).